### PR TITLE
Add InProcessAccuracyTest for Reach

### PR DESCRIPTION
**Pull request by renjiezh**

### CODE REVIEW

**Comment by wfa-reviewable** [comment]
This change is [Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1296)

File: src/main/kotlin/org/wfanet/measurement/integration/common/InProcessMeasurementReachAccuracyTest.kt [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+        RoLlv2ProtocolConfig.protocolConfig.sketchParams.maxSize,
+      )
+    val reachMeasurementParams =
+      ReachMeasurementParams(
+        StatsVidSamplingInterval(
+          measurementInfo.measurementSpec.vidSamplingInterval.start.toDouble(),
+          measurementInfo.measurementSpec.vidSamplingInterval.width.toDouble()
+        ),
+        DpParams(OUTPUT_DP_PARAMS.epsilon, OUTPUT_DP_PARAMS.delta),
+        StatsNoiseMechanism.GAUSSIAN
```
Comment (chenweiw) [inline src/main/kotlin/org/wfanet/measurement/integration/common/InProcessMeasurementReachAccuracyTest.kt:158]: Comparing to line 131, I am guessing StatsNoiseMechanism.Gaussian is for discrete Gaussian. Is it?

**Review by jiayu-google — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1296)* status: 0 of 8 files reviewed, 2 unresolved discussions (waiting on @chenweiw and @renjiezh)

___
*[`src/main/kotlin/org/wfanet/measurement/integration/common/InProcessMeasurementReachAccuracyTest.kt` line 244 at r2](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1296#-Nh96GKW-1vCVQtZQg8D:-Nh96GKW-1vCVQtZQg8E:b-esmzvx) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/6e6c1b85eaa41905fb04361bc1d362da30a62f29/src/main/kotlin/org/wfanet/measurement/integration/common/InProcessMeasurementReachAccuracyTest.kt#L244)):*
> ```Kotlin
> 
>     val standardDeviation = getStandardDeviation(reachResults.map { it.actualReach.toDouble() })
>     val standardDeviationOffset =
> ```

Hi Renjie, per discussion with @chenweiw , it's more forward-compatible to
```
assertTrue(standardDeviation > expectedStandardDeviation * VARIANCE_TEST_LOWER_THRESHOLD & standardDeviation > expectedStandardDeviation * VARIANCE_TEST_UPPER_THRESHOLD)
```
where `VARIANCE_TEST_LOWER_THRESHOLD` and `VARIANCE_TEST_UPPER_THRESHOLD` replace your current`VARIANCE_TEST_THRESHOLD`. You can set `VARIANCE_TEST_LOWER_THRESHOLD = 0.5` and `VARIANCE_TEST_UPPER_THRESHOLD = 1.5`, which is equivalent to `VARIANCE_TEST_THRESHOLD = 0.5` in your current codes.

**Review by renjiezh — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1296)* status: 0 of 8 files reviewed, 2 unresolved discussions (waiting on @chenweiw and @jiayu-google)

___
*[`src/main/kotlin/org/wfanet/measurement/integration/common/InProcessMeasurementReachAccuracyTest.kt` line 158 at r2](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1296#gh-1366201869:-Nh9EphF6n-gZbU9lNpc:b9if3ex) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/6e6c1b85eaa41905fb04361bc1d362da30a62f29/src/main/kotlin/org/wfanet/measurement/integration/common/InProcessMeasurementReachAccuracyTest.kt#L158)):*
<details><summary><i>Previously, chenweiw wrote…</i></summary><blockquote>

Comparing to line 131, I am guessing StatsNoiseMechanism.Gaussian is for discrete Gaussian. Is it?
</blockquote></details>

Yes, here it is discrete Gaussian.
On the other side, the noise mechanism in line 131 is used for direct measurement so that is only a placeholder. That parameter name is `expectedDirectNoiseMechanism`. 
___
*[`src/main/kotlin/org/wfanet/measurement/integration/common/InProcessMeasurementReachAccuracyTest.kt` line 244 at r2](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1296#-Nh96GKW-1vCVQtZQg8D:-Nh9FEawDBseB8ESZo4R:bu4ozqt) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/6e6c1b85eaa41905fb04361bc1d362da30a62f29/src/main/kotlin/org/wfanet/measurement/integration/common/InProcessMeasurementReachAccuracyTest.kt#L244)):*
<details><summary><i>Previously, jiayu-google wrote…</i></summary><blockquote>

Hi Renjie, per discussion with @chenweiw , it's more forward-compatible to
```
assertTrue(standardDeviation > expectedStandardDeviation * VARIANCE_TEST_LOWER_THRESHOLD & standardDeviation > expectedStandardDeviation * VARIANCE_TEST_UPPER_THRESHOLD)
```
where `VARIANCE_TEST_LOWER_THRESHOLD` and `VARIANCE_TEST_UPPER_THRESHOLD` replace your current`VARIANCE_TEST_THRESHOLD`. You can set `VARIANCE_TEST_LOWER_THRESHOLD = 0.5` and `VARIANCE_TEST_UPPER_THRESHOLD = 1.5`, which is equivalent to `VARIANCE_TEST_THRESHOLD = 0.5` in your current codes.
</blockquote></details>

Done.
Thanks!

**Review by chenweiw — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1296)* status: 0 of 8 files reviewed, 3 unresolved discussions (waiting on @jiayu-google, @renjiezh, and @SanjayVas)

___
*[`src/main/kotlin/org/wfanet/measurement/integration/common/InProcessMeasurementReachAccuracyTest.kt` line 244 at r2](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1296#-Nh96GKW-1vCVQtZQg8D:-NhDJKtp-nLKERjC3bAZ:b-jn2ptg) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/6e6c1b85eaa41905fb04361bc1d362da30a62f29/src/main/kotlin/org/wfanet/measurement/integration/common/InProcessMeasurementReachAccuracyTest.kt#L244)):*
<details><summary><i>Previously, renjiezh wrote…</i></summary><blockquote>

Done.
Thanks!
</blockquote></details>

In the line commented by Jiayu, the code after & should be "standardDeviation < ... '"

In addition, after discussing with Jiayu yesterday, we suggest to change standard deviation variance, i.e., 
```
assertTrue((standardDeviation).pow(2.0) > (expectedStandardDeviation).pow(2.0) * VARIANCE_TEST_LOWER_THRESHOLD & (standardDeviation).pow(2.0)  < (expectedStandardDeviation).pow(2.0)  * VARIANCE_TEST_UPPER_THRESHOLD)
``` 
or directly replace "(standard deviation).pow(2.0)" with "variance".  The threshold will be determined later.
___
*[`src/main/kotlin/org/wfanet/measurement/integration/common/InProcessMeasurementReachAccuracyTest.kt` line 271 at r3](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1296#-NhDKKqw4unanEr3wCb7:-NhDKKqw4unanEr3wCb8:b-eb4bj6) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/5a6b9752d0034dee7ec1aa5384204617878a8c07/src/main/kotlin/org/wfanet/measurement/integration/common/InProcessMeasurementReachAccuracyTest.kt#L271)):*
> ```Kotlin
>     private const val MULTIPLIER = 1.96
>     private const val CONTRIBUTOR_COUNT = 1
> ```

The four thresholds below for testing will be updated later.

**Review by SanjayVas — CHANGES_REQUESTED** [review]
> Reviewed 1 of 6 files at r1, 6 of 7 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1296)* status: 7 of 8 files reviewed, 7 unresolved discussions (waiting on @chenweiw, @jiayu-google, and @renjiezh)

___
*[`src/main/kotlin/org/wfanet/measurement/integration/common/InProcessCmmsComponents.kt` line 50 at r3](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1296#-NhDSfxm479srR4Yd4hF:-NhDSfxn9mXIX4jPRBCi:b2ni8l0) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/5a6b9752d0034dee7ec1aa5384204617878a8c07/src/main/kotlin/org/wfanet/measurement/integration/common/InProcessCmmsComponents.kt#L50)):*
> ```Kotlin
>   private val duchyDependenciesRule:
>     ProviderRule<(String, ComputationLogEntriesCoroutineStub) -> InProcessDuchy.DuchyDependencies>,
>   private val syntheticEventGroupSpec: List<SyntheticEventGroupSpec> =
> ```

nit



_Suggestion:_
```Kotlin
syntheticEventGroupSpecs
```
___
*[`src/main/kotlin/org/wfanet/measurement/integration/common/InProcessLifeOfAMeasurementIntegrationTest.kt` line 139 at r3](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1296#-NhDSwiSDaFUOdvksOGd:-NhDSwiT-cXD3_SOUAao:bhyd3iq) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/5a6b9752d0034dee7ec1aa5384204617878a8c07/src/main/kotlin/org/wfanet/measurement/integration/common/InProcessLifeOfAMeasurementIntegrationTest.kt#L139)):*
> ```Kotlin
>     runBlocking {
>       // Use frontend simulator to create a reach and frequency measurement and verify its result.
>       mcSimulator.testReachOnly("1234")
> ```

I'm having some trouble figuring out the difference in the name here. Technically all of the public methods in MeasurementConsumerSimulator could more accurately be renamed from "execute" to "test", since each does what a normal test method does (set up, call SUT, verify results).



_Code quote:_
```Kotlin
testReachOnly
```
___
*[`src/main/kotlin/org/wfanet/measurement/integration/common/SyntheticGenerationSpecs.kt` line 57 at r3](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1296#-NhDTeLn9hCn2lE0qjI_:-NhDTeLoDvZyHon4aimx:b-p7bi6m) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/5a6b9752d0034dee7ec1aa5384204617878a8c07/src/main/kotlin/org/wfanet/measurement/integration/common/SyntheticGenerationSpecs.kt#L57)):*
> ```Kotlin
>    * EventGroup specs for synthetic generation based on [POPULATION_SPEC].
>    *
>    * The total reach is ~2,000,000.
> ```

I assume this number is the value without any filters? The MC Simulator currently uses a hard-coded filter and date range.



_Code quote:_
```Kotlin
~2,000,000
```
___
*[`src/main/kotlin/org/wfanet/measurement/integration/common/InProcessMeasurementReachAccuracyTest.kt` line 60 at r3](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1296#-NhDTGbi1H4OeAVXmMI_:-NhDTGbi1H4OeAVXmMIa:bzmagv) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/5a6b9752d0034dee7ec1aa5384204617878a8c07/src/main/kotlin/org/wfanet/measurement/integration/common/InProcessMeasurementReachAccuracyTest.kt#L60)):*
> ```Kotlin
>  * easily.
>  */
> abstract class InProcessMeasurementReachAccuracyTest(
> ```

nit: maybe reverse the name to read more like standard English, since I assume this is specifically testing the REACH Measurement type.



_Suggestion:_
```Kotlin
InProcessReachMeasurementAccuracyTest
```

**Review by renjiezh — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1296)* status: 1 of 9 files reviewed, 4 unresolved discussions (waiting on @chenweiw, @jiayu-google, and @SanjayVas)

___
*[`src/main/kotlin/org/wfanet/measurement/integration/common/InProcessLifeOfAMeasurementIntegrationTest.kt` line 139 at r3](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1296#-NhDSwiSDaFUOdvksOGd:-NhQ-VgRDta2zftJ-nyS:bnsqdn7) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/5a6b9752d0034dee7ec1aa5384204617878a8c07/src/main/kotlin/org/wfanet/measurement/integration/common/InProcessLifeOfAMeasurementIntegrationTest.kt#L139)):*
<details><summary><i>Previously, SanjayVas (Sanjay Vasandani) wrote…</i></summary><blockquote>

I'm having some trouble figuring out the difference in the name here. Technically all of the public methods in MeasurementConsumerSimulator could more accurately be renamed from "execute" to "test", since each does what a normal test method does (set up, call SUT, verify results).
</blockquote></details>

Done.
Updated all function name to `testFoo`.
___
*[`src/main/kotlin/org/wfanet/measurement/integration/common/SyntheticGenerationSpecs.kt` line 57 at r3](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1296#-NhDTeLn9hCn2lE0qjI_:-NhPr2w22ggAII7_GQLu:b-wah6ax) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/5a6b9752d0034dee7ec1aa5384204617878a8c07/src/main/kotlin/org/wfanet/measurement/integration/common/SyntheticGenerationSpecs.kt#L57)):*
<details><summary><i>Previously, SanjayVas (Sanjay Vasandani) wrote…</i></summary><blockquote>

I assume this number is the value without any filters? The MC Simulator currently uses a hard-coded filter and date range.
</blockquote></details>

Yes. 
However, the MC Simulator's filter does not filter out anything for this spec.
___
*[`src/main/kotlin/org/wfanet/measurement/integration/common/InProcessMeasurementReachAccuracyTest.kt` line 244 at r2](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1296#-Nh96GKW-1vCVQtZQg8D:-NhQ-SbS2_7JYMCKBiQk:b-896fix) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/6e6c1b85eaa41905fb04361bc1d362da30a62f29/src/main/kotlin/org/wfanet/measurement/integration/common/InProcessMeasurementReachAccuracyTest.kt#L244)):*
<details><summary><i>Previously, chenweiw wrote…</i></summary><blockquote>

In the line commented by Jiayu, the code after & should be "standardDeviation < ... '"

In addition, after discussing with Jiayu yesterday, we suggest to change standard deviation variance, i.e., 
```
assertTrue((standardDeviation).pow(2.0) > (expectedStandardDeviation).pow(2.0) * VARIANCE_TEST_LOWER_THRESHOLD & (standardDeviation).pow(2.0)  < (expectedStandardDeviation).pow(2.0)  * VARIANCE_TEST_UPPER_THRESHOLD)
``` 
or directly replace "(standard deviation).pow(2.0)" with "variance".  The threshold will be determined later.
</blockquote></details>

Done.
___
*[`src/main/kotlin/org/wfanet/measurement/integration/common/InProcessMeasurementReachAccuracyTest.kt` line 271 at r3](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1296#-NhDKKqw4unanEr3wCb7:-NhPrs0d5oG9mytlsA_7:bhxjq64) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/5a6b9752d0034dee7ec1aa5384204617878a8c07/src/main/kotlin/org/wfanet/measurement/integration/common/InProcessMeasurementReachAccuracyTest.kt#L271)):*
<details><summary><i>Previously, chenweiw wrote…</i></summary><blockquote>

The four thresholds below for testing will be updated later.
</blockquote></details>

I see. Thanks.

**Review by SanjayVas — CHANGES_REQUESTED** [review]
> Reviewed 8 of 8 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1296)* status: all files reviewed, 5 unresolved discussions (waiting on @chenweiw, @jiayu-google, and @renjiezh)

___
*[`src/main/kotlin/org/wfanet/measurement/loadtest/measurementconsumer/MeasurementConsumerSimulator.kt` line 282 at r4](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1296#-NhSogYRESn3LDZj0nMv:-NhSogYRESn3LDZj0nMw:b-3sgjde) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/d53cf4e74cf2c3228504c034ef5547750b468190/src/main/kotlin/org/wfanet/measurement/loadtest/measurementconsumer/MeasurementConsumerSimulator.kt#L282)):*
> ```Kotlin
>   }
> 
>   suspend fun executeReachOnly(runId: String): ExecutionResult {
> ```

Make this private since callers are expected to call the "testFoo" methods only, in this case `testReachOnly`.
___
*[`src/main/kotlin/org/wfanet/measurement/integration/common/InProcessReachMeasurementAccuracyTest.kt` line 255 at r4](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1296#-NhSncY34riZV7HJ9Ui6:-NhSncY34riZV7HJ9Ui7:b-fp3tet) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/d53cf4e74cf2c3228504c034ef5547750b468190/src/main/kotlin/org/wfanet/measurement/integration/common/InProcessReachMeasurementAccuracyTest.kt#L255)):*
> ```Kotlin
>     )
> 
>     assertTrue(withinIntervalPercentage >= COVERAGE_TEST_THRESHOLD)
> ```

1. Use Truth for all assertions other than expected exceptions. e.g. `assertThat(withinIntervalPercentage).isAtLeast(AVERAGE_TEST_THRESHOLD)`
2. Use appropriate subjects, adding custom subjects as necessary. This means we get error message that are more helpful than "Expected 1 to be at least 2". See [FuzzyLongSubject](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/main/src/main/kotlin/org/wfanet/measurement/api/v2alpha/testing/FuzzyLongSubject.kt) and [MeasurementResultSubject](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/main/src/main/kotlin/org/wfanet/measurement/api/v2alpha/testing/MeasurementResultSubject.kt).



_Code quote:_
```Kotlin
assertTrue
```

**Review by renjiezh — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1296)* status: 8 of 9 files reviewed, 5 unresolved discussions (waiting on @chenweiw, @jiayu-google, and @SanjayVas)

___
*[`src/main/kotlin/org/wfanet/measurement/loadtest/measurementconsumer/MeasurementConsumerSimulator.kt` line 282 at r4](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1296#-NhSogYRESn3LDZj0nMv:-NhTFIMP0iQKH4ckgRwV:baoq1fk) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/d53cf4e74cf2c3228504c034ef5547750b468190/src/main/kotlin/org/wfanet/measurement/loadtest/measurementconsumer/MeasurementConsumerSimulator.kt#L282)):*
<details><summary><i>Previously, SanjayVas (Sanjay Vasandani) wrote…</i></summary><blockquote>

Make this private since callers are expected to call the "testFoo" methods only, in this case `testReachOnly`.
</blockquote></details>

`executeReachOnly()` is also called by the AccuracyTest. This is the reason I extracted it out.
___
*[`src/main/kotlin/org/wfanet/measurement/integration/common/InProcessReachMeasurementAccuracyTest.kt` line 255 at r4](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1296#-NhSncY34riZV7HJ9Ui6:-NhTFF7mBsslK6nEXM5o:b38t97z) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/d53cf4e74cf2c3228504c034ef5547750b468190/src/main/kotlin/org/wfanet/measurement/integration/common/InProcessReachMeasurementAccuracyTest.kt#L255)):*
<details><summary><i>Previously, SanjayVas (Sanjay Vasandani) wrote…</i></summary><blockquote>

1. Use Truth for all assertions other than expected exceptions. e.g. `assertThat(withinIntervalPercentage).isAtLeast(AVERAGE_TEST_THRESHOLD)`
2. Use appropriate subjects, adding custom subjects as necessary. This means we get error message that are more helpful than "Expected 1 to be at least 2". See [FuzzyLongSubject](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/main/src/main/kotlin/org/wfanet/measurement/api/v2alpha/testing/FuzzyLongSubject.kt) and [MeasurementResultSubject](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/main/src/main/kotlin/org/wfanet/measurement/api/v2alpha/testing/MeasurementResultSubject.kt).
</blockquote></details>

Done.
Thanks for the explanation.

**Review by SanjayVas — APPROVED** [review]
> Reviewed 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1296)* status: all files reviewed, 3 unresolved discussions (waiting on @chenweiw and @jiayu-google)

**Review by Marco-Premier — APPROVED** [review]
> Reviewed 1 of 7 files at r2, 7 of 8 files at r4, 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1296)* status: all files reviewed, 3 unresolved discussions (waiting on @chenweiw and @jiayu-google)

**Review by chenweiw — CHANGES_REQUESTED** [review]
The thresholds for the coverage test and the mean test have not been updated. I included the two comments in the review. All the other changes look good.

> Reviewed 1 of 6 files at r1, 6 of 7 files at r2, 1 of 1 files at r3, 7 of 8 files at r4, 2 of 2 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1296#-:-NiaOyo66IN09YnzsIsq:bornmds)* status: all files reviewed, 3 unresolved discussions (waiting on @jiayu-google and @renjiezh)

___
*[`src/main/kotlin/org/wfanet/measurement/integration/common/InProcessReachMeasurementAccuracyTest.kt` line 267 at r7](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1296#-NiaMXbY6qYW2sQHdiSt:-NiaMXbY6qYW2sQHdiSu:brk97j3) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/be8a694fe00610ad93839874eca7ca72239fa065/src/main/kotlin/org/wfanet/measurement/integration/common/InProcessReachMeasurementAccuracyTest.kt#L267)):*
> ```Kotlin
>     private const val MULTIPLIER = 1.96
> 
>     private const val COVERAGE_TEST_THRESHOLD = 90
> ```

I recommend 80 as COVERAGE_TEST_THRESHOLD.
___
*[`src/main/kotlin/org/wfanet/measurement/integration/common/InProcessReachMeasurementAccuracyTest.kt` line 268 at r7](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1296#-NiaNpJw49NAoh1Zebsr:-NiaNpJw49NAoh1Zebss:bz80mqn) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/be8a694fe00610ad93839874eca7ca72239fa065/src/main/kotlin/org/wfanet/measurement/integration/common/InProcessReachMeasurementAccuracyTest.kt#L268)):*
> ```Kotlin
> 
>     private const val COVERAGE_TEST_THRESHOLD = 90
>     private const val AVERAGE_TEST_THRESHOLD = 4
> ```

I recommend 2.58 as AVERAGE_TEST_THRESHOLD.

**Review by renjiezh — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1296)* status: 8 of 11 files reviewed, 3 unresolved discussions (waiting on @chenweiw, @jiayu-google, @Marco-Premier, and @SanjayVas)

___
*[`src/main/kotlin/org/wfanet/measurement/integration/common/InProcessReachMeasurementAccuracyTest.kt` line 267 at r7](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1296#-NiaMXbY6qYW2sQHdiSt:-NiaQuppDiY--O__MDMS:b-896fix) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/be8a694fe00610ad93839874eca7ca72239fa065/src/main/kotlin/org/wfanet/measurement/integration/common/InProcessReachMeasurementAccuracyTest.kt#L267)):*
<details><summary><i>Previously, chenweiw wrote…</i></summary><blockquote>

I recommend 80 as COVERAGE_TEST_THRESHOLD.
</blockquote></details>

Done.
___
*[`src/main/kotlin/org/wfanet/measurement/integration/common/InProcessReachMeasurementAccuracyTest.kt` line 268 at r7](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1296#-NiaNpJw49NAoh1Zebsr:-NiaQuDKBqN9LCydeUQ1:b-896fix) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/be8a694fe00610ad93839874eca7ca72239fa065/src/main/kotlin/org/wfanet/measurement/integration/common/InProcessReachMeasurementAccuracyTest.kt#L268)):*
<details><summary><i>Previously, chenweiw wrote…</i></summary><blockquote>

I recommend 2.58 as AVERAGE_TEST_THRESHOLD.
</blockquote></details>

Done.

**Review by chenweiw — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1296)* status: 8 of 11 files reviewed, 1 unresolved discussion (waiting on @jiayu-google, @Marco-Premier, and @SanjayVas)

**Review by jiayu-google — APPROVED** [review]
:lgtm: Thank you Renjie!

> Reviewed 1 of 7 files at r2, 6 of 8 files at r4, 1 of 2 files at r7, 2 of 3 files at r8, 1 of 1 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1296#-:-Nib3wNNDThf3Uc9eJ3P:b7mrrhs)* status: all files reviewed, 1 unresolved discussion (waiting on @SanjayVas)

**Review by renjiezh — COMMENTED** [review]
> Reviewed 1 of 7 files at r2, 6 of 8 files at r4, 1 of 2 files at r7, 2 of 3 files at r8, 1 of 1 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1296)* status: all files reviewed, 1 unresolved discussion (waiting on @SanjayVas)

**Review by renjiezh — COMMENTED** [review]
> Dismissed @jiayu-google from a discussion.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1296)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @SanjayVas)

**Pull request merged**